src/c++/perf_analyzer/raw_data_collector.h

```diff
@@ -0,0 +1,105 @@
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+
+#include <map>
+#include <tuple>
+
+#include "client_backend/client_backend.h"
+#include "constants.h"
+#include "perf_utils.h"
+#include "request_record.h"
+
+namespace triton { namespace perfanalyzer {
+
+struct PerfMode {
+  uint32_t concurrency;
+  double request_rate;
+
+  PerfMode()
+  {
+    concurrency = 0;
+    request_rate = 0.0;
+  }
+
+  PerfMode(uint64_t c, double rr)
+  {
+    concurrency = c;
+    request_rate = rr;
+  }
+
+  bool operator==(const PerfMode& rhs) const
+  {
+    return (concurrency == rhs.concurrency) &&
+           (request_rate == rhs.request_rate);
+  }
+};
+
+struct Experiment {
+  PerfMode mode;
+  std::vector<RequestRecord> requests;
+  std::vector<uint64_t> window_boundaries;
+};
+
+class RawDataCollector {
+ public:
+  static cb::Error Create(std::shared_ptr<RawDataCollector>* collector);
+  ~RawDataCollector() = default;
+
+
+  /// Add a measurement window to the collector
+  /// \param id Identifier for the experiment
+  /// \param window_start_ns The window start timestamp in nanoseconds.
+  /// \param window_end_ns The window end timestamp in nanoseconds.
+  void AddWindow(
+      PerfMode& id, uint64_t window_start_ns, uint64_t window_end_ns);
+
+  /// Add request records to an experiment
+  /// @param id Identifier for the experiment
+  /// @param request_records The request information for the current experiment.
+  void AddData(PerfMode& id, std::vector<RequestRecord>&& request_records);
+
+  /// Get the raw experiment data for the profile
+  /// @return Raw experiment data
+  std::vector<Experiment>& GetData() { return experiments_; }
+
+  std::string& GetVersion() { return version_; }
+
+ private:
+  RawDataCollector() = default;
+
+  std::vector<Experiment>::iterator FindEntry(PerfMode& id)
```

Comment:
FindExperiment? Is that what it is doing?

Reply:
Yes. Renamed to `FindExperiment` 👍